lib/pkgcloud/azure-v2/compute/client/servers.js

```diff
@@ -143,71 +149,110 @@ function createServer(options, callback) {
 
   if (!options.name || !options.username || !options.password) {
     return errs.handle(
-      errs.create({ message: 'Please provide a name for the vm, as well as the username and password for login' }),
+      errs.create({
+        message: 'Please provide a name for the vm, as well as the username and password for login'
+      }),
       callback
     );
   }
 
   if (!options.flavor) {
     return errs.handle(
-      errs.create({ message: 'When creating an azure server a flavor or an image need to be supplied' }),
+      errs.create({
+        message: 'When creating an azure server a flavor or an image need to be supplied'
+      }),
       callback
     );
   }
 
-  var adjustVMTemplate = function (template) {
-
-    var vmIndex = _.findIndex(template.resources, { 'type': 'Microsoft.Compute/virtualMachines' });
-
-    // Adding additional data disks
-    if (options.storageDataDiskNames && options.storageDataDiskNames.length) {
-      options.storageDataDiskNames.forEach(function (ddName, idx) {
-        template.resources[vmIndex].properties.storageProfile.dataDisks.push({
-          'name': 'datadisk' + idx.toString(),
-          'diskSizeGB': '100',
-          'lun': 0,
-          'vhd': {
-            'uri': '[concat(reference(concat(\'Microsoft.Storage/storageAccounts/\', variables(\'storageAccountName\')), \'2016-01-01\').primaryEndpoints.blob, parameters(\'storageContainerName\'),\'/\', \'' + ddName + '\', \'.vhd\')]'
-          },
-          'createOption': 'Empty'
-        });
-      });
```

Comment:
Why did you remove those lines?
Those lines enable adding multiple data disks when creating/updating a vm